### ISSUE TITLE
Some questions about training my custom datasets

### ISSUE THREAD

**Issue by bjutlss** [Microsoft/singleshotpose#185]
First of all, thank you for your work!
First，I consulted ObjectDatasetsTools and made my custom datasets with D435i.
1.I read this paper and learned that the regularization parameter needs to be set to 0 in pre-training. So am I right to train my data set directly with the following code?
`python train.py --datacfg cfg/myobj.data --modelcfg cfg/yolo-pose.cfg --initweightfile cfg/darknet19_448.conv.23 --pretrain_num_epochs 15`
or
`python train.py --datacfg cfg/myobj.data --modelcfg cfg/yolo-pose.cfg --initweightfile backup/duck/init.weights`
What's the difference between yolo-pose.cfg and yolo-pose-pre. cfg?
yolo -pose-pre. cfg is used for pre-training my target object?

2.After training, I had ran valid.py to evaluate the model and get the following results:
![image](https://user-images.githubusercontent.com/102348012/219602741-3f20b52c-96e7-4c4a-a137-0deeaaf42120.png)
The Acc 3D transformation is significantly lower than the other two terms. Is this a normal result?

3.If the above steps are correct, I take a frame of the video stream and use it as RGB input to get the results of R_pr and t_pr, which coordinate system are they relative to the camera? (I used Realsense D435i).I'm going to view the resulting RT matrix as relative to the optical coordinate system. It was sent to the robot arm and got an incorrect grab.

@btekin @snsinha  Can you answer these questions for me? Look forward to your quick answer！

**Comment by AristideLaignel** [comment]
Hi @bjutlss , 

I have no answer to your question, but I'm trying to train the model on the LINEMOD dataset and I have no good results. 

How did you do to train your model and with which version of CUDA, python and PyTorch ?

I'm using CUDA 11.6, python 3.9.16, pytorch 1.13.1
and with this command :

![command](https://user-images.githubusercontent.com/46279890/225879340-3689925b-a345-4b66-8003-f6432d14456a.png)

and I have this result : 

![iron_train](https://user-images.githubusercontent.com/46279890/225877791-f67ee037-4a08-449c-9320-24eedac39b1e.png)

and these parameters in yolo-pose.cfg : 

![yolo-pose cfg](https://user-images.githubusercontent.com/46279890/225877874-9738359c-7823-4db4-989c-1355d6708a57.png)

Did you know which parameters to adjust to have a good result on the training ?

Many thanks 

Aristide

**Comment by bjutlss** [comment]
Well,&nbsp; My CUDA 11.1, python 3.8, pytorch 1.8.0 .
I just read the README to train the model without changing any parameters except batchsize.
But,&nbsp;The problem could be the wrong 3D model or the wrong diameter parameter.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/singleshotpose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月17日(星期五) 晚上6:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





 
Hi @bjutlss ,
 
I have no answer to your question, but I'm trying to train the model on the LINEMOD dataset and I have no good results.
 
How did you do to train your model and with which version of CUDA, python and PyTorch ?
 
I'm using CUDA 11.6, python 3.9.16, pytorch 1.13.1
 and with this command :
 

 
and I have this result :
 

 
and these parameters in yolo-pose.cfg :
 

 
Did you know which parameters to adjust to have a good result on the training ?
 
Many thanks
 
Aristide

**Comment by AristideLaignel** [comment]
Thanks you for your quick answer !
I will try with your version of pytorch and python.

**Comment by bjutlss** [comment]
If you can train normally, then it shouldn't be a problem with the CUDA, python version, the 3D models and&nbsp; rgb、labels don't correspond and that can happen.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/singleshotpose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月17日(星期五) 晚上6:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





 
Thanks you for your quick answer !
 I will try with your version of pytorch and python.

**Comment by AristideLaignel** [comment]
So why I have no results on the training with the exact parameters, I change nothing and the validation works fine with there weights but not the training, I don't understand !!
Here what I have on all images, it seems to detect but not with the right dimensions : 

![1AAABOX3D9](https://user-images.githubusercontent.com/46279890/225888971-036f47bd-c402-4fb9-be09-ac81feb88cd7.png)

![1AAABOX3D10](https://user-images.githubusercontent.com/46279890/225888992-85636c36-41a7-44e7-ba6c-03b05b330e6e.png)

![1AAABOX3D1](https://user-images.githubusercontent.com/46279890/225889588-c96e830c-cba4-4c13-b6c7-7f1eab05c790.png)


![1AAABOX3D0](https://user-images.githubusercontent.com/46279890/225889623-436fdcac-3464-4fc7-bd5d-18c0df09ed5f.png)

![1AAABOX3D79](https://user-images.githubusercontent.com/46279890/225889651-b04aa3fb-cb81-4dc3-94f1-4dc4543da10b.png)


The prediction is in red and the ground truth in green

**Comment by bjutlss** [comment]
You may have to find out for yourself, but that's all I can help you with. I just said what I thought!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/singleshotpose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月17日(星期五) 晚上7:12
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





 
So why I have no results on the training with the exact parameters, I change nothing and the validation works fine with there weights but not the training, I don't understand !!
 Here what I have on all images, it seems to detect but not with the right dimensions :
 

 ![1AAABOX3D10](https://user-images.githubusercontent.com/46279890/225888992-85636c36-41a7-44e7-ba6c-03b05b330e6
 
 e.png)
 The prediction is in red and the ground truth in green

**Comment by AristideLaignel** [comment]
Many thanks for your response. 
Just what do you have in your yolo-pose.cfg about weight and height please ? 
thanks

**Comment by bjutlss** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/singleshotpose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月17日(星期五) 晚上7:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





 
Many thanks for your response.
 Just what do you have in your yolo-pose.cfg about weight and height please ?
 thanks

**Comment by AristideLaignel** [comment]
Sorry I can't see your response correctly

**Comment by bjutlss** [comment]
[net]
# io
batch=4
height=416
width=416
channels=3
num_keypoints=9

# training
momentum=0.9
decay=0.0005
angle=0
burn_in=1000
max_batches = 80200
policy=steps
max_epochs=500
learning_rate=0.001
steps=-1,80,160
scales=0.1,0.1,0.1

# test - eliminate low confidence predictions during testing
conf_thresh= 0.1
test_width=672
test_height=672

# data augmentation
saturation = 1.5
exposure = 1.5
hue=.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky


#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=64
activation=leaky

[reorg]
stride=2

[route]
layers=-1,-4

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
# filters=125
filters=20
activation=linear


[region]
anchors = 
bias_match=1
classes=1
coords=18
num=1
softmax=1
jitter=.3
rescore=1

object_scale=5
noobject_scale=0.1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/singleshotpose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月17日(星期五) 晚上9:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





 
Sorry I can't see your response correctly

**Comment by AristideLaignel** [comment]
Thank you @bjutlss I have the same !

**Comment by zongt-yu** [comment]
> 谢谢@bjutlss我有同样的 ！

I have the same problem, did you solve it?

**Comment by bjutlss** [comment]
Not yet




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月25日(星期六) 下午2:10
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [microsoft/singleshotpose] Some questions about training my custom datasets (Issue #185)





  
***@***.***我有同样的 ！
  
I have the same problem, did you solve it?
